tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/if-stmt.bal

```diff
@@ -217,6 +217,47 @@ function testResetTypeNarrowingWithBlockStmt() {
     }
 }
 
+const DOT = ".";
+
+public function testTypeNarrowingWithReturningIf1() returns error? {
+    int i = 0;
+    string word = "";
+    while i < 3 {
+        string:Char? char = check getChar();
+        if char is () {
+            return error("Failed");
+        } else if char !is DOT {
+            return error("Failed");
+        }
+        word += char;
+        i = i + 1;
+    }
+
+    assertEquality("...", word);
+}
+
+public function testTypeNarrowingWithReturningIf2() returns error? {
+    int i = 0;
+    string word = "";
+    while i < 3 {
+        string:Char? char = check getChar();
+        if char is () {
+            return error("Failed");
+        } else if char !is DOT {
+            return error("Failed");
+        }
+        i = i + 1;
+        assertTrue(word is DOT);
+        word += char;
+    }
+
+    assertEquality("...", word);
+}
+
+public function getChar() returns string:Char|error? {
+    return check string:fromCodePointInt(89);
+}
+
```

Comment:
Shall we add a negative test like below
```
public function main() returns error? {
    string|int|float? char = "a";
    string word = "";
    if char is () {
        return error("char is null");
    } else if char is int {
        return error("char is int");
    } else if char == "a" {

    }

    word += char; 
}
```

To show that type of `char` is narrowed to `string|float` properly

Reply:
Also add below positive case as well
```
public function main() returns error? {
    string|int|float? char = "a";
    string word = "";
    if char is () {
        return error("char is null");
    } else if char is int {
        return error("char is int");
    }  else if char is float {
        return error("char is float");
    } 

    word += char; 
}
```